### PR TITLE
Feedback Implementer from Manager & Peers

### PR DESCRIPTION
**Pull request by Calebc4554**
# Description  
- **What does this PR do?**  
  This PR implements several improvements based on peer mentor and manager feedback.
  Key updates include:  
  - Fallback messaging on the leaderboard when fewer than 3 users exist  
  - `submitting` state to prevent duplicate form submissions  
  - Updated UI labels for clarity (“Years of Experience”)  
  - Unified `loading` state handling across multiple components  

- **Why is it necessary or valuable?**  
  This PR addresses usability and UI issues raised during code reviews and improves the user experience. 

- **What is intentionally left out and will be done in a later PR?**  
  - Multi-role selection for mentors/mentees  
  - More nuanced preferred meeting type (e.g., "Email", "Chat")  
  - Generalized year ranges for experience (stretch improvement)

---

# Resources and References  
- **Internal references:**  
  - Feedback from @alegleason, @Sijia11 @weibinma123 on GitHub PR threads  
---

# Test Plan  
- **How was this tested?**  
  - Verified leaderboard fallback appears with <3 users  
  - Created test events to confirm submission prevention  
  - Updated profile fields and verified rendering

## Edge Cases Tested  
- Leaderboard with <3 users  
- Submitting event multiple times rapidly  
- Profiles with no experience/skills/interests  

---

# Additional Notes  
- **Known limitations:**  
  - Still using choice options instead of multi-select for meeting type and roles  
  - Experience year ranges are still numeric (not bracketed yet)

### CODE REVIEW

File: frontend/src/components/Events.jsx [hunk 1]
```diff
@@ -12,6 +12,7 @@ export default function Events({ role }) {
 	const [points, setPoints] = useState(0);
 	const [loading, setLoading] = useState(true);
 	const [showModal, setShowModal] = useState(false);
+	const [submitting, setsSubmitting] = useState(false);
```
Comment (Sijia11) [inline frontend/src/components/Events.jsx:15]: Is this a typo, it should be setSubmitting?

Comment (Calebc4554) [reply frontend/src/components/Events.jsx:15]: Yes thanks!
